mkosi/__init__.py

```diff
@@ -143,6 +144,11 @@ def install_distribution(state: MkosiState) -> None:
             with umask(~0o500):
                 (state.root / "efi").mkdir(exist_ok=True)
 
+            # Some distributions install EFI binaries directly to /boot/efi. Let's redirect them to /efi
+            # instead.
+            rmtree(state.root / "boot/efi")
+            (state.root / "boot/efi").symlink_to("../efi")
```

Comment:
Well this is a can of worms.

So currently, when an ESP is added, systemd-repart will exclude the contents of /boot and /efi when populating the root partition. Also, when populating a VFAT partition, systemd-repart will skip any symlinks it finds. So if a user adds an XBOOTLDR, it will contain just the contents of /boot without the /boot/efi symlink. This will go horribly wrong when trying to update stuff via the package manager, but then again, that's already the case as it is